core/src/client/async_client/mod.rs

```diff
@@ -726,6 +622,38 @@ async fn background_task<S, R>(
 		}
 	}
 
+	let res = loop {
+		tokio::select! {
+			maybe_rx = rx.recv() => {
+				let Some(msg) = maybe_rx else {
+					break Ok(())
+				};
+
+				if let Err(e) = sender.send(msg).await {
+					break Err(Error::Transport(e.into()));
+				}
+			}
+			_ = ping_fut(ping_interval) => {
+				if let Err(e) = sender.send_ping().await {
+					break Err(Error::Transport(e.into()));
+				}
+			}
+			_ = close_tx.closed() => {
+				break Ok(())
```

Comment:
Could some tracing debug logs be useful on the break branches to know the reason of the `send_task` exit?

Reply:
I added `Client dropped` logs when` either the send or receive task returns `Ok(())` ok?

These channels should be closed if send/receive tasks are closed, if something terminates with an error that much occur before this and we don't add confusing logs.

However, if it's somewhere that could be tricky detect by looking at the logs.